uaparser/deprecated.go

```diff
@@ -0,0 +1,57 @@
+package uaparser
+
+import (
+	"os"
+
+	"gopkg.in/yaml.v3"
+)
+
+// NewWithOptions is deprecated.
+// Deprecated: Use New and option functions instead.
+func NewWithOptions(regexFile string, mode, treshold, topCnt int, useSort, debugMode bool, cacheSize int) (*Parser, error) {
+	uaRegexes, err := os.ReadFile(regexFile)
+	if err != nil {
+		return nil, err
+	}
+
+	var def *RegexesDefinitions
+
+	if err := yaml.Unmarshal(uaRegexes, def); err != nil {
+		return nil, err
+	}
+
+	return New(
+		WithRegexesDefinitions(def),
+		WithMode(LookupMode(mode)),
+		WithMissesThreshold(uint64(treshold)),
+		WithMatchIdxNotOk(topCnt),
+		WithSort(useSort),
+		WithDebug(debugMode),
+		WithCacheSize(cacheSize),
+	)
+}
+
+// NewFromSaved is deprecated.
+// Deprecated: Use New and option functions instead.
```

Comment:
```suggestion
// Deprecated: Use New() instead.
```
behavior of this function is almost no different than behavior of `New()` with no options, per the implementation below.